src/eradiate/test_tools/regression.py

```diff
@@ -54,50 +63,172 @@ def regression_test_plots(data: xr.Dataset, filename: PathLike, metric: tuple) -
     plt.close()
 
 
+def reference_converter():
+    """
+    A converter for handling the reference data attribute.
+
+    The produced converter processes a parameter ``value`` as follows:
+
+    * if ``value`` is a string or path like object, it attempts to load
+      a dataset from that location and returns ``None`` if that fails.
+    * if ``value`` is a xarray dataset, it is returned directly.
+
+    Returns
+    -------
+    callable
+        Generated converter.
+    """
+
+    def f(value):
+        if isinstance(value, (str, os.PathLike)):
+            if not os.path.isfile(value):
+                return None
+            return xr.load_dataset(value)
+        elif isinstance(value, xr.Dataset):
+            return value
+
+    return f
```

Comment:
You still don't need to return a callable:
```suggestion
def reference_converter(value):
    if isinstance(value, (str, os.PathLike)):
        if not os.path.isfile(value):
            return None
        return xr.load_dataset(value)
    elif isinstance(value, xr.Dataset):
        return value
```
What happens if `value` is neither a path-like nor a dataset?